### ISSUE TITLE
go-blockdevice has wrong checksum

### ISSUE THREAD

**Issue by nberlee** [siderolabs/talos#5807]
## Bug Report
As an Arch user, I compile talosctl using the AUR. This fails with 1.1.0 because the go-blockdevice module has a checksum mismatch.

### Logs
```
verifying github.com/talos-systems/go-blockdevice@v0.3.2: checksum mismatch
	downloaded: h1:uzPwAxb+4RZIoDGlGMeUBqRd9++Z2IGLGwnxAh8atS0=
	/home/nicob/.cache/yay/talosctl/src/talos/go.sum:     h1:qa966a7JH6ORv7xUFb1JyBjXAEHqTxhfCzU0ARzUKek=

SECURITY ERROR
This download does NOT match an earlier download recorded in go.sum.
The bits may have been replaced on the origin server, or an attacker may
have intercepted the download attempt.
```

So I did a manual independent check:
```
$ go install  github.com/vikyd/go-checksum@latest
go: downloading github.com/vikyd/go-checksum v1.0.0
$  git clone https://github.com/siderolabs/go-blockdevice -b v0.3.2 
Cloning into 'go-blockdevice'...
[..]
Note: switching to 'fccee8bb082b105cb60db40cb01636efc3241b5f'.
[..]
$ ~/go/bin/go-checksum go-blockdevice/  github.com/talos-systems/go-blockdevice@v0.3.2
directory: go-blockdevice/
{
	"HashSynthesized": "bb33f00316fee11648a031a518c79406a45df7ef99d8818b1b09f1021f1ab52d",
	"HashSynthesizedBase64": "uzPwAxb+4RZIoDGlGMeUBqRd9++Z2IGLGwnxAh8atS0=",
	"GoCheckSum": "h1:uzPwAxb+4RZIoDGlGMeUBqRd9++Z2IGLGwnxAh8atS0="
}
```

**Comment by smira** [comment]
I'm confused here, as the tag hasn't changed.

**Comment by smira** [comment]
And I can't reproduce wrong checksum:

```bash
$ docker run --rm -it golang:1.18-alpine
/go/aaaa # go mod init github.com/smira/aa
go: creating new go.mod: module github.com/smira/aa
/go/aaaa # go get github.com/talos-systems/go-blockdevice@v0.3.2
go: downloading github.com/talos-systems/go-blockdevice v0.3.2
go: added github.com/talos-systems/go-blockdevice v0.3.2
/go/aaaa # cat go.sum
github.com/armon/circbuf v0.0.0-20190214190532-5111143e8da2/go.mod h1:3U/XgcO3hCbHZ8TKRvWD2dDTCfh9M9ya+I9JpbB7O8o=
github.com/creack/pty v1.1.9/go.mod h1:oKZEueFk5CKHvIhNR5MUki03XCEU+Q6VDXinZuGJ33E=
github.com/davecgh/go-spew v1.1.0/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
github.com/davecgh/go-spew v1.1.1/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
github.com/google/uuid v1.1.2/go.mod h1:TIyPZe4MgqvfeYDBFedMoGGpEw/LqOeaOT+nhxU+yHo=
github.com/kr/pty v1.1.1/go.mod h1:pFQYn66WHrOpPYNljwOMqo10TkYh1fy3cYio2l3bCsQ=
github.com/kr/text v0.1.0/go.mod h1:4Jbv+DJW3UT/LiOwJeYQe1efqtUx/iVham/4vfdArNI=
github.com/kr/text v0.2.0/go.mod h1:eLer722TekiGuMkidMxC/pM04lWEeraHUUmBw8l2grE=
github.com/niemeyer/pretty v0.0.0-20200227124842-a10e7caefd8e/go.mod h1:zD1mROLANZcx1PVRCS0qkT7pwLkGfwJo4zjcN/Tysno=
github.com/pmezard/go-difflib v1.0.0/go.mod h1:iKH77koFhYxTK1pcRnkKkqfTogsbg7gZNVY4sRDYZ/4=
github.com/ryanuber/go-glob v1.0.0/go.mod h1:807d1WSdnB0XRJzKNil9Om6lcp/3a0v4qIHxIXzX/Yc=
github.com/stretchr/objx v0.1.0/go.mod h1:HFkY916IF+rwdDfMAkV7OtwuqBVzrE8GR6GFx+wExME=
github.com/stretchr/testify v1.7.0/go.mod h1:6Fq8oRcR53rry900zMqJjRRixrwX3KX962/h/Wwjteg=
github.com/talos-systems/go-blockdevice v0.3.2 h1:qa966a7JH6ORv7xUFb1JyBjXAEHqTxhfCzU0ARzUKek=
github.com/talos-systems/go-blockdevice v0.3.2/go.mod h1:qnn/zDc09I1DA2BUDDCOSA2D0P8pIDjN8pGiRoRaQig=
github.com/talos-systems/go-cmd v0.0.0-20210216164758-68eb0067e0f0/go.mod h1:kf+rZzTEmlDiYQ6ulslvRONnKLQH8x83TowltGMhO+k=
github.com/talos-systems/go-retry v0.1.1-0.20201113203059-8c63d290a688/go.mod h1:HiXQqyVStZ35uSY/MTLWVvQVmC3lIW2MS5VdDaMtoKM=
golang.org/x/sys v0.0.0-20201130171929-760e229fe7c5/go.mod h1:h1NjWce9XRLGQEsW7wpKNCjG9DtNlClVuFLEZdDNbEs=
golang.org/x/text v0.3.4/go.mod h1:5Zoc/QRtKVWzQhOtBMvqHzDpF6irO9z98xDceosuGiQ=
golang.org/x/tools v0.0.0-20180917221912-90fa682c2a6e/go.mod h1:n7NCudcB/nEzxVGmLbDWY5pfWTLqBcC2KZ6jyYvM4mQ=
gopkg.in/check.v1 v0.0.0-20161208181325-20d25e280405/go.mod h1:Co6ibVJAznAaIkqp8huTwlJQCZ016jof/cbN4VW5Yz0=
gopkg.in/check.v1 v1.0.0-20200902074654-038fdea0a05b/go.mod h1:Co6ibVJAznAaIkqp8huTwlJQCZ016jof/cbN4VW5Yz0=
gopkg.in/yaml.v3 v3.0.0-20200313102051-9f266ea9e77c/go.mod h1:K4uyk7z7BCEPqu6E+C64Yfv1cQ7kz7rIZviUmN+EgEM=
gopkg.in/yaml.v3 v3.0.0-20200615113413-eeeca48fe776/go.mod h1:K4uyk7z7BCEPqu6E+C64Yfv1cQ7kz7rIZviUmN+EgEM=
```

**Comment by frezbo** [comment]
I remember someone else also seeing this in arch due to the way it builds or something

**Comment by smira** [comment]
There should be something with arch, I wonder if it's Go version or something, but it doesn't look to be on our side so far.

**Comment by nberlee** [comment]
Thats weird, super weird.. Will look in to it.

Using a container I can build it, without it, just a plain `go build cmd/talosctl` makes it fail on the checksum.

Now for the super weird part. Whenever I calculate the checksum using `go-checksum` it I get the same `uzP..` checksum on arch and in a golang:1.18-alpine or golang:1.18 (debian) container.

I will look in to it further... As for the arch part. It yes it was also broken in Talosctl 1.0 because `github.com/docker/distribution` retagged that version (and talos had the old hash, see https://github.com/distribution/distribution/issues/3590)

**Comment by nberlee** [comment]
Yay, I am not going crazy, nor is my FS broken

@smira I managed to break it using your testing method:
```
docker run --rm -it golang:1.18-alpine
/go # cd /
/ # mkdir src
/ # cd src
/src # export GOPROXY=direct
/src # go mod init github.com/smira/aa
go: creating new go.mod: module github.com/smira/aa
/src # apk add git
fetch https://dl-cdn.alpinelinux.org/alpine/v3.16/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.16/community/x86_64/APKINDEX.tar.gz
(1/6) Installing brotli-libs (1.0.9-r6)
(2/6) Installing nghttp2-libs (1.47.0-r0)
(3/6) Installing libcurl (7.83.1-r1)
(4/6) Installing expat (2.4.8-r0)
(5/6) Installing pcre2 (10.40-r0)
(6/6) Installing git (2.36.1-r0)
Executing busybox-1.35.0-r13.trigger
OK: 19 MiB in 21 packages
/src # go get github.com/talos-systems/go-blockdevice@v0.3.2
go: downloading github.com/talos-systems/go-blockdevice v0.3.2
go: github.com/talos-systems/go-blockdevice@v0.3.2: verifying module: checksum mismatch
	downloaded: h1:uzPwAxb+4RZIoDGlGMeUBqRd9++Z2IGLGwnxAh8atS0=
	sum.golang.org: h1:qa966a7JH6ORv7xUFb1JyBjXAEHqTxhfCzU0ARzUKek=

SECURITY ERROR
This download does NOT match the one reported by the checksum server.
The bits may have been replaced on the origin server, or an attacker may
have intercepted the download attempt.

For more information, see 'go help module-auth'.
```
Please note the `export GOPROXY=direct`

This means is downloads it from github, and not from proxy.golang.org. 
So the problem is, there is a cached mismatch between the golang proxy server and the github source. This is nasty to say the least!

And matches my findings. The github v0.3.2 branch has a different hash when compared to the proxied one 
 golang.org/lookup/github.com/talos-systems/go-blockdevice@v0.3.2

**Comment by smira** [comment]
thanks, fixed via #5822, we'll get it into v1.1.1
